src/Microsoft.AspNetCore.SignalR.Redis/RedisHubLifetimeManager.cs

```diff
@@ -241,11 +281,22 @@ public override Task OnDisconnectedAsync(HubConnectionContext connection)
 
         public override async Task AddGroupAsync(string connectionId, string groupName)
         {
+            if (await AddGroupAsyncCore(connectionId, groupName))
+            {
+                // short circuit if connection is on this server
+                return;
+            }
+
+            await SendGroupActionAndWaitForAck(connectionId, groupName, GroupAction.Add);
+        }
+
+        private async Task<bool> AddGroupAsyncCore(string connectionId, string groupName)
+        {
             var groupChannel = _channelNamePrefix + ".group." + groupName;
```

Comment:
nit: this should be moved closer to the usage. We are doing this but may not use at all if the connection with the given id does not exist (same in RemoveGroupAsync)